subsys/bluetooth/host/hci_core.c

```diff
@@ -262,13 +263,23 @@ void bt_send_one_host_num_completed_packets(uint16_t handle)
 	BT_ASSERT_MSG(err == 0, "Unable to send Host NCP (err %d)", err);
 }
 
+#if defined(CONFIG_BT_TESTING)
+__weak void bt_testing_trace_event_acl_pool_destroy(struct net_buf *buf)
```

Comment:
Seems like we might need a `include/zephyr/bluetooth/testing.h` which would contain the various prototypes of testing-related functions and also properly document them. I noticed we have some existing cases that could use it, like here:
https://github.com/zephyrproject-rtos/zephyr/blob/1471c2f81430fd0a381c37d263a4d0ff966e0cb4/tests/bsim/bluetooth/host/misc/disconnect/dut/src/main.c#L32-L33

Reply:
Do we want to expose test functions in a public header file? 

Honestly not sure what the best practices is for testing in Zephyr/C when it comes to things like this

Reply:
Seems like that concern should be solvable with proper documentation. These APIs either way require explicitly enabling `CONFIG_BT_TESTING` before they can be used. We could additionally output a build warning (if we don't already) that the configuration should not be used in production.

Reply:
@alwa-nordic this conversation is still unresolved

Reply:
I welcome the "public" test header. Should I do add it?

Should its doxygen say `@file @internal`? That will prevent anything from that file from rendering in the public documentation, I think.

Reply:
I'm not a fan of public testing headers. If this is for internal testing, then I think such headers are better placed in a non-public place. Placing a `testing.h` makes more sense if it's a testing API that is designed to be used for users (maybe that is the intention?). 

We actually recently moved the Mesh `testing.h` header file from the public include directory to internal.

Reply:
The main issue with putting them e.g. somewhere under `subsys/bluetooth/` is the then you need to play manual tricks with include lookup paths etc. An alternative idea that comes to mind is to have some `CONFIG_BT_PRIVATE_INCLUDES` option which adds `subsys/bluetooth/` to the lookup path.

Reply:
Semantically it's odd to have an `internal` directory in the `public` directory :D 

FYI I'm not blocking these changes as I don't have a solution/alternative, but just wanted to voice my concerns about putting functions that aren't meant to be used by applications in the same location as our APIs. We don't want it to be perceived as a test framework/interface of any sort, right?